aten/src/ATen/native/special_functions/prime_number.h

```diff
@@ -0,0 +1,1029 @@
+#pragma once
+
+#include <ATen/native/special_functions/detail/promote.h>
+#include <c10/macros/Macros.h>
+
+namespace at {
+namespace native {
+namespace special_functions {
+template<typename T1>
+C10_HOST_DEVICE
+inline constexpr
+T1
+prime_number(unsigned int n) {
+  constexpr std::uint16_t prime_numbers[10000] = {
```

Comment:
Hmm, this would create an instance of this array for every instantiation of this function, isn't it?
I.e. wouldn't it be better to just define it in prime_numers namespace? (though I'm not sure how it will work with C10_HOST_DEVICE decorator)